be/src/vec/columns/column_array.h

```diff
@@ -45,6 +45,18 @@ class ColumnArray final : public COWHelper<IColumn, ColumnArray> {
 
     ColumnArray(const ColumnArray&) = default;
 
+public:
+    // offsets of array is 64bit wise
+    using Offset64 = IColumn::Offset64;
+    using Offsets64 = IColumn::Offsets64;
+
+private:
+    // please use IColumn::Offset if we really need 32bit offset, otherwise use ColumnArray::Offset64
+    using Offset [[deprecated("ColumnArray::Offset64 for Array, IColumn::Offset for String")]] =
```

Comment:
Why should we still keep the ColumnArray::Offset, are there any other place still using it?

Reply:
Do not find any method to delete typedef Offset in ColumnArray now.

So we define it inside private region and mark it as deprecated.